src/arch/src/lib.rs

```diff
@@ -63,20 +63,20 @@ impl From<GuestMemoryError> for Error {
 pub type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Default)]
-pub struct FdtBuilder<'a> {
-    cmdline: Option<&'a str>,
+pub struct FdtBuilder {
+    cmdline: Option<String>,
```

Comment:
When doing the squash, please leave this change in a separate commit as it is only mildly related to the rest of the changes. You should add an explanation in the commit message why was this change added.